diesel/src/expression/expression_methods/global_expression_methods.rs

```diff
@@ -295,11 +295,10 @@ pub trait ExpressionMethods: Expression + Sized {
     ///     use self::posts::dsl::{posts, author_name};
     ///     let connection = establish_connection();
     ///
-    ///     let data = users.inner_join(posts)
+    ///     let data: QueryResult<Vec<String>> = users.inner_join(posts)
     ///         .filter(name.nullable().eq(author_name))
```

Comment:
One function per line is much easier to parse than:

```
let data: QueryResult<Vec<String>> = users.inner_join(posts)
    .filter(name.nullable().eq(author_name)).select(name).load(&connection)
    .map(Iterator::collect);
```

Did you add the comment to this line to address the filter() function specifically?

I don't think we need to separate this to be multiline though:
```
.filter(
    name.nullable()
        .eq(author_name)
)
```